### PR TITLE
add firebase in-app messaging display pod

### PR DESCRIPTION
**Pull request by mylyuic**
This is the firebase in-app messaging display pod, which is the default display implementation for fiam.  It requires firebase in-app messaging of version 0.12.0 or higher.

### CODE REVIEW

File: Firebase/InAppMessagingDisplay/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FirebaseInAppMessagingDisplay is the default UI implementation from Firebase for
+rendering In-App Messaging messges to end users.
+
+Apps can also provide custom UI implementation to replace it. Check out our guides
```
Comment (paulb777) [inline Firebase/InAppMessagingDisplay/README.md]: Add link to guides - then lgtm

Comment (paulb777) [reply Firebase/InAppMessagingDisplay/README.md]: sgtm

**Comment by paulb777** [comment]
Please wait for @ryanwilson to review before merging.

File: Firebase/InAppMessagingDisplay/FIRIAMDefaultDisplayImpl.m [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
++ (void)displayModalViewWithMessageDefinition:(FIRInAppMessagingModalDisplay *)modalMessage
+                              displayDelegate:
+                                  (id<FIRInAppMessagingDisplayDelegate>)displayDelegate {
+  NSBundle *resourceBundle = [self getViewResourceBundle];
+
+  if (resourceBundle == nil) {
+    NSError *error = [NSError errorWithDomain:kFirebaseInAppMessagingDisplayErrorDomain
+                                         code:FIAMDisplayRenderErrorTypeUnspecifiedError
+                                     userInfo:@{@"message" : @"resource bundle is missing"}];
```
Comment (ryanwilson) [inline Firebase/InAppMessagingDisplay/FIRIAMDefaultDisplayImpl.m]: Does this get presented to the user? Or just the developer?

File: Firebase/InAppMessagingDisplay/FIRCore+InAppMessagingDisplay.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+#import <FirebaseCore/FIRLogger.h>
+
+// This file contains declarations that should go into FirebaseCore when
+// Firebase InAppMessagingDisplay is merged into master. Keep them separate now to help
+// with build from development folder and avoid merge conflicts.
+
+// this should eventually be in FIRLogger.h
```
[the remark this answers is not in the document]
Comment (paulb777) [reply Firebase/InAppMessagingDisplay/FIRCore+InAppMessagingDisplay.h]: Right, we want to move the service constants to the clients

File: Firebase/InAppMessagingDisplay/Banner/FIDBannerViewController.m [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+@property(weak, nonatomic)
+    IBOutlet NSLayoutConstraint *imageBottomAlignWithBodyLabelBottomConstraint;
+@property(weak, nonatomic) IBOutlet UIImageView *imageView;
+@property(weak, nonatomic) IBOutlet UILabel *titleLabel;
+@property(weak, nonatomic) IBOutlet UILabel *bodyLabel;
+
+// Banner view will be rendered and dismissed with animation. Within viewDidLayoutSubviews function,
+// we would position the view so that it's out of UIWindow range on the top so that later on it can
+// be slided in with animation. However, viewDidLayoutSubviews is also triggred in other scenarios
```
[the remark this answers is not in the document]
Comment (mylyuic) [reply Firebase/InAppMessagingDisplay/Banner/FIDBannerViewController.m]: done

File: Firebase/InAppMessagingDisplay/Banner/FIDBannerViewController.m [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+  UITapGestureRecognizer *tapGestureRecognizer =
+      [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(messageTapped:)];
+  tapGestureRecognizer.delaysTouchesBegan = YES;
+  tapGestureRecognizer.numberOfTapsRequired = 1;
+
+  [self.view addGestureRecognizer:tapGestureRecognizer];
+}
+
+static const CGFloat kSwipeUpThreshold = -10.0f;
```
[the remark this answers is not in the document]
Comment (mylyuic) [reply Firebase/InAppMessagingDisplay/Banner/FIDBannerViewController.m]: moved

File: Firebase/InAppMessagingDisplay/FIDBaseRenderingViewController.m [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+- (void)followActionURL {
+  [self.view.window setHidden:YES];
+  // This is for the purpose of releasing the potential memory associated with the image view.
+  self.view.window.rootViewController = nil;
+
+  if (self.displayDelegate) {
+    [self.displayDelegate messageClicked];
+  } else {
+    FIRLogWarning(kFIRLoggerInAppMessagingDisplay, @"I-FID200008", @"Display delegate is nil");
```
[the remark this answers is not in the document]
Comment (mylyuic) [reply Firebase/InAppMessagingDisplay/FIDBaseRenderingViewController.m]: Changed the text.

File: Firebase/InAppMessagingDisplay/FIDRenderingWindowHelper.h [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * To avoid the risk of hijacking the app's existing view transition flow, we render in-app message
+ * views in a top level UI Window instead of presenting from app's existing UIWindow. The caller is
+ * supposed to set the rootViewController to be the appropriate view controller for the in-app
+ * message and call setHidden:NO to make it really visible.
+ */
+@interface FIDRenderingWindowHelper : NSObject
+
+// Return the singleton UIWindow that can be used for rendering modal IAM views
++ (UIWindow *)uiWindowForModalView;
```
[the remark this answers is not in the document]
Comment (mylyuic) [reply Firebase/InAppMessagingDisplay/FIDRenderingWindowHelper.h]: changed to UIWindow

File: Firebase/InAppMessagingDisplay/FIRCore+InAppMessagingDisplay.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (mylyuic) [reply Firebase/InAppMessagingDisplay/FIRCore+InAppMessagingDisplay.h]: removed the comment

**Pull request merged**